DECOMPRESSING.md

```diff
@@ -0,0 +1,34 @@
+# Decompressing code.bin files
+This is a tutorial on how to get, extract, and decompress code.bin files for patch compilation.
+
+## Method 1
+1. Have a hacked 3DS
+2. Open GodMode9, then open "Title manager" -> "NAND / TWL"
+3. Find all modules you need, then on each one, press A, then press "Manage Title...", then press "Build CIA (standard)".
+4. Get all the CIAs (located in sd:/gm9/out) onto your computer in a dedicated folder using ftpd or an SD card reader.
+5. Place a ctrtool executable in that folder.
+6. Now run `ctrtool --contents=contents cianame.cia` (replace cianame with the cia's name), then run `ctrtool --exefs exefs.bin contentsname` (replace contentsname with the name of the file the previous command created), then last but not least, run `ctrtool --exefsdir=exefs --decompersscode exefs.bin`
+7. Move the code.bin folder into its proper patches folder, then delete the exefs.bin, contents file, and the CIA for that module if you would like.
+8. Repeat steps 6-7 for the rest of your CIAs (or just make a script to do it for you)
+9. Done!
+
+## Method 2
+1. Have a hacked 3DS
+2. Open GodMode9, then open "Title manager" -> "NAND / TWL"
+3. Find a module you need, then press A, then press "Open title folder".
+4. Then press A on the .app file in the folder, select "NCCH image options...".
+5. Select "Extract .code"
+6. 0:/gm9/out should now have a file called <Titleid>.dec.code, which is your code.bin for that app.
+7. Repeat steps 3-6 for the remaining modules.
+8. Done!
+
+## Method 3 (Requires external programs)
+1. Have a hacked 3DS
+2. Download dump_modules.gm9 from [here](https://github.com/TraceEntertains/nimbus-module-extractor/releases/latest)
+3. Put it in sd:/gm9/scripts
+4. Run the script (Home Button -> Scripts -> dump_modules.gm9)
+5. Turn off your 3DS and remove the SD card.
+6. Put the SD card in your PC.
+7. Extract the zip with the exe in it from [here](https://github.com/TraceEntertains/nimbus-module-extractor/releases/latest) (same link again)
+8. Drag and drop your nimbus folder onto the executable while your 3DS SD card is in your PC.
+9. Done!
```

Comment:
Yeah, definitely needs some changes, and method 1 is kinda useless, but I will defend method 3, because it took me a good few hours to make it and is less tedious than manually extracting/decompressing the modules.
Will get it updated when I get a chance.

Reply:
I understand that you spent time on method 3, but you also have to understand my point of view. "I spent time on it" is not exactly a good justification for why a feature should be added, it should be based on what objective improvements it brings to the codebase. Having a single tool/script that dumps everything at once _**is**_ a good idea, but method 3 does not do that. Method 3 only dumps the entire exefs from the titles, and then still requires processing on the users machine after being dumped. To me that negates any time save gained from not going through each title, since going through each title gets you usable code.bin files immediately. It's also only useful for Windows users, since you only provide an extractor build for Windows. I'm also not sure what exactly this extractor brings to the table that ctrtool doesn't already do?

Reply:
> For method 3, you can use the [extrcode](https://github.com/d0k3/GodMode9/blob/b7c97af144f84fcd89de43421e97d1ab7bacdb74/resources/sample/HelloScript.gm9#L280) command on the GodMode9 script, which will extract the code.bin from the app file directly

This was replied as I was typing mine, didn't see it until now. This is the best option overall, and would solve the issues I had with method 3. @DaniElectra Is this the decompressed code.bin or is it the compressed version?

Reply:
> > For method 3, you can use the [extrcode](https://github.com/d0k3/GodMode9/blob/b7c97af144f84fcd89de43421e97d1ab7bacdb74/resources/sample/HelloScript.gm9#L280) command on the GodMode9 script, which will extract the code.bin from the app file directly
> 
> This was replied as I was typing mine, didn't see it until now. This is the best option overall, and would solve the issues I had with method 3. @DaniElectra Is this the decompressed code.bin or is it the compressed version?

I sent the comment on the review, and you posted the comment before I completed it. Yes, it's the decompressed code.

Reply:
Perfect, _that's_ the solution that should be used then

Reply:
Ok, will use that in the next script. Should I make it to where the structure of the stored files matches nimbus so you can just merge, or rely on the user to move the files to the right spots?

Reply:
I would match the structure with nimbus to make things easier

Reply:
The script has been updated, just released 2.0 which uses the new method.